src/lib.rs

```diff
@@ -752,6 +752,92 @@ impl Rand for XorShiftRng {
     }
 }
 
+/// A xoroshiro+ random number generator.
+///
+/// The xoroshiro+ algorithm is not suitable for cryptographic purposes, but
+/// is very fast and has better statistical properties than `XorShiftRng`.  If
+/// you do not know for sure that it fits your requirements, use a more secure
+/// one such as `IsaacRng` or `OsRng`.
+///
+/// The algorithm used here is translated from [the `xoroshiro128plus.c`
+/// reference source code](http://xorshift.di.unimi.it/xoroshiro128plus.c) by
+/// David Blackman and Sebastiano Vigna.
+#[allow(missing_copy_implementations)]
+#[derive(Clone)]
+pub struct XoroShiroRng {
+    s0: u64,
+    s1: u64,
+}
+
+impl XoroShiroRng {
+    /// Creates a new XoroShiroRng instance which is not seeded.
+    ///
+    /// The initial values of this RNG are constants, so all generators created
+    /// by this function will yield the same stream of random numbers. It is
+    /// highly recommended that this is created through `SeedableRng` instead of
+    /// this function
+    pub fn new_unseeded() -> XoroShiroRng {
+        // These constants were taken from the XorShiftRng implementation above.
+        // The only requirement imposed by the algorithm is that these values
+        // cannot be zero everywhere.
+        XoroShiroRng {
+            s0: 0x193a6754a8a7d469,
+            s1: 0x97830e05113ba7bb,
+        }
+    }
+}
+
+impl Rng for XoroShiroRng {
+    #[inline]
+    fn next_u32(&mut self) -> u32 {
+        self.next_u64() as u32
+    }
```

Comment:
Would it make sense to reuse the second half of the 64-bit integer if a second 32-bit integer is requested? Or is this slower due to the additional branch?